### DISCUSSION TITLE
How to use parallel computing/multithreading on ISIS4

### DISCUSSION THREAD

**Discussion by pruthviacharya** [DOI-USGS/ISIS3#5226] [General]
I have ISIS 4.3.1 installed on a Linux machine with 10 cores and 20 threads; however, when I am running cam2map and it is taking forever. According to System Monitor on my system, only one CPU core is getting used at a time.

Is there an option that I can use to enable multithreading or use parallel computing to speed up the process.

**Comment by jlaura** [comment]
@pruthviacharya `cam2map` is (unfortunately) neither threaded nor capable of parallel processing. The classic technique to apply is to use an application like [GNU Parallel](https://www.gnu.org/software/parallel/) to run many concurrent `cam2map` calls on different input observations at once.

Depending on your use case, you might consider using the [ASP `mapproject`](https://stereopipeline.readthedocs.io/en/latest/tools/mapproject.html) application that tiles the data and can be run using multiple processes.

**Comment by pruthviacharya** [comment]
I just need to do a polar projection for a MARCI image. I will check out `mapproject`. 

Thanks!

**Comment by thareUSGS** [comment]
You might also want to check out this script (the bash *.sh file) which shows how one can use GNU Parallel. This codes happens to be using MARCI also. Looks like it is only issuing 2-6 parallel calls but that can be as many as you want. In your case you might want to limit it to ~10.
https://github.com/AsterAphelion/marci-parallel/tree/main
